packages/harpparts/src/apparatus/utils/map-hole-to-include-valvebends/map-hole-to-include-valvebends.test.ts

```diff
@@ -0,0 +1,91 @@
+import type { Hole } from '../../types'
+
+import { mapHoleToIncludeValvebends } from './map-hole-to-include-valvebends'
+
+test('mapHoleToIncludeValvebends adds a valved blow bend when the blow reed is lower', () => {
+  const holeInput: Hole = {
+    blow: 9,
+    draw: 10,
+    blowbends: [],
+    drawbends: [],
+    overblows: [],
+    overdraws: [],
+    valvedblows: [],
+    valveddraws: [],
+  }
+
+  const expectedHoleOutput: Hole = {
+    ...holeInput,
+    valvedblows: [8],
+  }
+  const actualHoleOutput = mapHoleToIncludeValvebends(holeInput)
+
+  expect(actualHoleOutput).toStrictEqual(expectedHoleOutput)
+})
+
+test('mapHoleToIncludeValvebends adds a valved draw bend when draw reed is lower', () => {
+  const holeInput: Hole = {
+    blow: 10,
+    draw: 9,
+    blowbends: [],
+    drawbends: [],
+    overblows: [],
+    overdraws: [],
+    valvedblows: [],
+    valveddraws: [],
+  }
+
+  const expectedHoleOutput: Hole = {
+    ...holeInput,
+    valveddraws: [8],
+  }
+  const actualHoleOutput = mapHoleToIncludeValvebends(holeInput)
+
+  expect(actualHoleOutput).toStrictEqual(expectedHoleOutput)
+})
+
+// The way I see it, if you're halfvalving normally but your hole would have no bends
+// for the unvalved reed then you might as well go full valved.
+// This is an edge case anyway.
```

Comment:
This needs rewording a bit, otherwise it would seem to justify going full valved on holes where the reeds are only a semitone apart. It doesn't because actually, while there isn't a full bend there there is a microtonal one which *is* useful and probably preferable to valved bends in most cases because it's kinder on the reed and easier to control the amplitude.

Reply:
Done